packages/amplify-cli-core/src/types.ts

```diff
@@ -20,19 +20,209 @@ export type $TSContext = {
    * @deprecated
    */
   print: IContextPrint;
-  migrationInfo: $TSAny;
+  migrationInfo: MigrationInfo;
   projectHasMobileHubResources: boolean;
   prompt: $TSAny;
   exeInfo: $TSAny;
-  input: $TSAny;
-  parameters: $TSAny;
-  usageData: $TSAny;
-  runtime: $TSAny;
+  input: $CommandLineInput;
+  parameters: ContextParameters;
+  usageData: IUsageData;
+  runtime: Runtime;
   pluginPlatform: IPluginPlatform;
-  newUserInfo?: $TSAny;
+  newUserInfo?: string;
   filesystem: IContextFilesystem;
   template: IContextTemplate;
-  updatingAuth: $TSAny;
+  updatingAuth: any; // this seems to actually implement CognitoStackOptions
+};
+
+export interface MigrationInfo {
+  amplifyMeta: $TSMeta;
+  newVersion: string;
+  initVersion: string;
+  currentAmplifyMeta: $TSMeta;
+  projectConfig: ProjectConfig;
+  projectPath: string;
+  localEnvInfo: LocalEnvInfo;
+  localAwsInfo: LocalAwsInfo;
+  teamProviderInfo: TeamProviderInfo;
+  backendConfig: Record<string, unknown>;
+}
+export type TeamProviderEnvironment = {
+  categories: Record<string, unknown>;
+};
+export type TeamProviderInfo = {
+  [envName: string]: Record<string, unknown>;
+};
+export type LocalAwsInfo = {
+  NONE: unknown;
+};
```

Comment:
what does this do ?

Reply:
https://github.com/sdstolworthy/amplify-cli/blob/0f94a470070a5e4ff6f2b4a1212f9f6a81e0fbd6/packages/amplify-cli/src/migrate-project.ts#L243

It's not clear to me. But the best I could tell was that this value in the context of MigrationInfo will always return `{ NONE: <contents of awsInfo.json>}`.

Reply:
seems that
```
export type LocalAwsInfo = {
  NONE: Record<string, unknown>;
};
```
would be more appropriate here.